httputil/httputil.go

```diff
@@ -267,18 +267,20 @@ func ListenAndServe(opts ...Option) error {
 			errs <- (&http.Server{
 				ReadTimeout:  5 * time.Second,
 				WriteTimeout: 5 * time.Second,
-				Handler: http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-					w.Header().Set("Connection", "close")
-					url := "https://" + req.Host + req.URL.String()
-					http.Redirect(w, req, url, http.StatusMovedPermanently)
-				}),
+				Handler:      m.HTTPHandler(http.HandlerFunc(redirect)),
```

Comment:
I think I'll need to refactor this a little since currently this code will only run if `redirectHTTPS` is true and we might be using Let's Encrypt but not doing redirects in that case these changes wouldn't work.

Reply:
Added a fixup in 178158e. If redirectHTTPS is true or we've set the GetCertificate config which indicates we're using Let's Encrypt then we will start the http listener. Using the redirect methods in both cases should be fine because the autocert.HTTPHandler will do a redirect anyway if nil is passed in which is the other option we could do. I think this should correctly work now in all cases